### PR TITLE
Dialback SMS sending

### PR DESCRIPTION
**Pull request by NicolaasDieBaas**
Should be applied in multiple registration lines.

### CODE REVIEW

File: go-app-ussd_clinic.js [hunk 1]
```diff
@@ -115,6 +146,14 @@ go.app = function() {
                 registrant_info.details.source = "clinic";
             }
 
+            if (registrant_info.details.clinic) {
+                registrant_info.details.clinic.redial_sms_sent = self.im.user.answers.redial_sms_sent;
+            } else {
+                registrant_info.details.clinic = {};
+                registrant_info.details.clinic.redial_sms_sent = self.im.user.answers.redial_sms_sent;
```
Comment (gsvr) [inline go-app-ussd_clinic.js]: You could just put this in the dict in the line above to make it a bit clearer, i.e.
`registrant_info.details.clinic = {redial_sms_sent: self.im.user.answers.redial_sms_send}`

**Comment by NicolaasDieBaas** [comment]
@gsvr please review

File: test/ussd_public.test.js [hunk 2]
```diff
@@ -148,6 +148,73 @@ describe("app", function() {
...
+                    , "1"  // state_registered_full - compliment
+                    , {session_event: 'close'}
+                )
+                .check.user.answer("redial_sms_sent", true)
+                .check(function(api) {
+                    utils.check_fixtures_used(api, [53, 54, 127, 187]);
+                })
+                .run();
+            });
+            it("updates identity with redial_sms_sent 'false'", function() {
```
Comment (gsvr) [inline test/ussd_public.test.js]: Change this description to `should not send dial-back sms when timeout occurs on a non-dialback state` or something similar.

File: test/ussd_chw.test.js [hunk 3]
```diff
@@ -169,6 +169,59 @@ describe("app", function() {
         });
         // end re-dial flow tests
 
+        describe("dialback sms testing", function() {
+            it("send if redial sms not yet sent", function() {
```
Comment (gsvr) [inline test/ussd_chw.test.js]: I'd like to see explicit testing for when an identity loads with and without the `chw.redial_sms_sent` field, and similar for other apps. You don't need to complete the whole flow, just this part.

**Comment by NicolaasDieBaas** [comment]
@gsvr another review please

File: test/ussd_clinic.test.js [hunk 4]
```diff
@@ -209,7 +209,22 @@ describe("app", function() {
                 })
                 .run();
             });
-            it("updates identity with redial_sms_sent 'true'", function() {
+            it("send if redial sms not yet sent (identity loads without redial_sms_sent set as 'false')", function() {
```
Comment (gsvr) [inline test/ussd_clinic.test.js]: `without` it set as 'false'? that seems the same as the first test.

File: test/ussd_public.test.js [hunk 5]
```diff
@@ -170,7 +170,21 @@ describe("app", function() {
                 })
                 .run();
             });
-            it("updates identity with redial_sms_sent 'false'", function() {
+            it("send if redial sms not yet sent (identity loads without redial_sms_sent set as 'false')", function() {
```
Comment (gsvr) [inline test/ussd_public.test.js]: same here with `without`

**Review by gsvr — CHANGES_REQUESTED** [review]

File: test/ussd_clinic.test.js (hunk 4 above)
Comment (NicolaasDieBaas) [reply test/ussd_clinic.test.js]: 'without' should be 'with'... i'll make the changes

**Comment by NicolaasDieBaas** [comment]
@gsvr yet another review please

File: test/ussd_chw.test.js [hunk 6]
```diff
@@ -169,6 +163,74 @@ describe("app", function() {
...
+                )
+                .check.user.answer("redial_sms_sent", true)
+                .check(function(api) {
+                    utils.check_fixtures_used(api, [185]);
+                })
+                .run();
+            });
+            it("send if redial sms not yet sent (identity loads with redial_sms_sent set as 'false')", function() {
+                return tester
+                .setup.user.addr("27820001008")
```
Comment (gsvr) [inline test/ussd_chw.test.js:196]: This identity only has it set to `false` for `nurseconnect` - it should be `false` for all the lines you're testing it against.

**Review by gsvr — CHANGES_REQUESTED** [review]

File: test/ussd_chw.test.js (hunk 6 above)
Comment (NicolaasDieBaas) [reply test/ussd_chw.test.js:196]: oops yes!! sorry.  will change

**Comment by NicolaasDieBaas** [comment]
@gsvr review please

**Review by gsvr — APPROVED** [review]

**Pull request merged**